pkg/config/seelog_config.go

```diff
@@ -0,0 +1,92 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+package config
+
+import (
+	"bytes"
+	"strings"
+	"sync"
+	"text/template"
+)
+
+// SeelogConfig abstract seelog XML configuration definition
+type SeelogConfig struct {
+	settings map[string]interface{}
+	m        sync.Mutex
+}
+
+const seelogConfigurationTemplate = `
+<seelog minlevel="{{.logLevel}}">
+	<outputs formatid="{{.format}}">
+    {{if .consoleLoggingEnabled   }}<console />{{end}}
+    {{if .logfile                 }}<rollingfile type="size" filename="{{.logfile}}" maxsize="{{.maxsize}}" maxrolls="{{.maxrolls}}" />{{end}}
+	{{if .syslogURI               }}<custom name="syslog" formatid="syslog-{{.format}}" data-uri="{{.syslogURI}}" data-tls="{{.syslogUseTLS}}" />{{end}}
+	</outputs>
+	<formats>
+		<format id="json" format="{{.jsonFormat}}"/>
+		<format id="common" format="{{.commonFormat}}"/>
+		<format id="syslog-json" format="%CustomSyslogHeader(20,{{.syslogRFC}}){&quot;agent&quot;:&quot;{{.loggerName | ToLower}}&quot;,&quot;level&quot;:&quot;%LEVEL&quot;,&quot;relfile&quot;:&quot;%ShortFilePath&quot;,&quot;line&quot;:&quot;%Line&quot;,&quot;msg&quot;:&quot;%Msg&quot;}%n"/>
```

Comment:
This is a bit creepy and hard to read, isn't there a way to do this with `template/html`? That looks wrong to me that we have to manually escape all these chars (but maybe there is no other way).

Reply:
I totally agree that embedding xml this way is terrible, still slightly better than before. But I will give it a try with `html/template` and hopefully remove html escaping sequence like `&quot;` thus making the whole big string slightly better.

Reply:
TBH my comment was mainly about the `&quot;` part, embedding the XML this way is not ideal but will definitely do the job.

Reply:
I reworked the templating logic, most of the json-inside-xml escaping constraints are offloaded to `html/template` and overall I think it makes the whole stuff (slightly) easier to read.

Reply:
Awesome!